spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/CommitOwnerClientImplSuiteBase.scala

```diff
@@ -198,15 +198,19 @@ trait CommitOwnerClientImplSuiteBase extends QueryTest
       val log = DeltaLog.forTable(spark, tempDir.toString)
       val logPath = log.logPath
       val tableCommitOwnerClient = createTableCommitOwnerClient(log)
-      tableCommitOwnerClient.commitOwnerClient.registerTable(
-        logPath, currentVersion = -1L, initMetadata, Protocol(1, 1))
 
       val e = intercept[CommitFailedException] {
         commit(version = 0, timestamp = 0, tableCommitOwnerClient)
       }
       assert(e.getMessage === "Commit version 0 must go via filesystem.")
       writeCommitZero(logPath)
-      assert(tableCommitOwnerClient.getCommits() == GetCommitsResponse(Seq.empty, -1))
+      var expectedVersion = -1
+      if (tableCommitOwnerClient.commitOwnerClient.isInstanceOf[DynamoDBCommitOwnerClient]) {
+        // DynamoDBCommitOwnerClient stores attemptVersion as the current table version when
```

Comment:
`DynamoDBCommitOwnerClient.getCommits` starts advertising `attemptVersion` as the current commit version even without knowledge around whether the commit has gone through or not. Practically this won't cause any issue because until the commit is done, no client will invoke `DynamoDBCommitOwnerClient.getCommits` since the table was a FS table. But just for the sake of semantics, we should fix it in a followup PR.

Reply:
Yes, for DynamoDB, we treat registerTable as a valid commit and store the attemptVersion instead of the current Delta table version. This is not risky because the DynamoDB table entry is only treated as valid if the corresponding filesystem commit goes through (which means that the attemptVersion is in fact the Delta table version). In case of a filesystem conflict, it is expected that registerTable will be called again. However, for consistency, we can use another bool to track whether any commits have gone through DynamoDB and return -1.